pkg/image/manifest.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// NOTE: This is manually replicated from: https://github.com/kubernetes/kubernetes/blob/master/test/utils/image/manifest.go
+
+package image
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	version "github.com/hashicorp/go-version"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// RegistryList holds public and private image registries
+type RegistryList struct {
+	DockerLibraryRegistry string `yaml:"dockerLibraryRegistry"`
+	E2eRegistry           string `yaml:"e2eRegistry"`
+	EtcdRegistry          string `yaml:"etcdRegistry"`
+	GcRegistry            string `yaml:"gcRegistry"`
+	PrivateRegistry       string `yaml:"privateRegistry"`
+	SampleRegistry        string `yaml:"sampleRegistry"`
+
+	K8sVersion *version.Version
+	Images     map[int]Config
+}
+
+// Config holds an images registry, name, and version
+type Config struct {
+	registry string
+	name     string
+	version  string
+}
+
+// NewRegistryList returns a default registry or one that matches a config file passed
+func NewRegistryList(repoConfig, k8sVersion string) (*RegistryList, error) {
+	registry := &RegistryList{
+		DockerLibraryRegistry: "docker.io/library",
+		E2eRegistry:           "gcr.io/kubernetes-e2e-test-images",
+		EtcdRegistry:          "quay.io/coreos",
+		GcRegistry:            "k8s.gcr.io",
+		PrivateRegistry:       "gcr.io/k8s-authenticated-test",
+		SampleRegistry:        "gcr.io/google-samples",
+	}
+
+	// Load in a config file
+	if repoConfig != "" {
+
+		fileContent, err := ioutil.ReadFile(repoConfig)
+		if err != nil {
+			panic(fmt.Errorf("Error reading '%v' file contents: %v", repoConfig, err))
```

Comment:
I know this was copied from the upstream, but since we are using this logic within Sonobuoy I think we should avoid the panic (which upstream in testing is OK, on a CLI we shouldn't).

I'd change these panics to print the errors and exit 1.